### PR TITLE
A11y: Fix canvas element accessibility

### LINKED ISSUE #70974: A11y: Fix keyboard accessibility in `public/app/features/canvas/runtime/element.tsx`
# How to make progress on this issue

- 🔍 open the file and search for `TODO: fix keyboard a11y`
- ❌ remove the `eslint-disable` comment
- 🔧 fix the lint error(s) that should now be showing
  - this will be heavily dependant on context
  - should you be able to interact with this element via the keyboard or is there another element that already handles the same behaviour via the keyboard?
  - in some cases, the best fix is to **use the correct semantic element** for the job. if something behaves like a button but we want it to look different, we can still use the `<button type="button">` element and style it appropriately. this will handle all our keyboard accessibility needs for us. there's even a handy [`clearButtonStyles`](https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/Button/Button.tsx#L306) util to undo the basic button styling.
  - if using a `<button>` or other similar interactive element is not appropriate, refer to the documentation [here](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/click-events-have-key-events.md) and [here](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/no-static-element-interactions.md). this will likely involve setting a `role` and implementing a custom `onKeyPress`
- 🧪 test it!
  - can you interact with this element using the keyboard now?
- 🥳 celebrate for making grafana a better product!

### PR DESCRIPTION
**Pull request by Develer**
**What is this feature?**

Fix canvas element accessibility.

**Which issue(s) does this PR fix?**:

Fixes #70974 

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/canvas/runtime/element.tsx [hunk 1]
```diff
@@ -486,6 +486,24 @@ export class ElementState implements LayerElement {
...
+      if (scene?.tooltipCallback && scene.tooltip?.anchorPoint) {
+        const rect = this.div?.getBoundingClientRect();
+        scene.tooltipCallback({
+          anchorPoint: { x: rect?.right ?? scene.tooltip.anchorPoint.x, y: rect?.top ?? scene.tooltip.anchorPoint.y },
+          element: this,
+          isOpen: true,
+        });
+      }
+    }
+  };
```
Comment (nmarrs) [inline public/app/features/canvas/runtime/element.tsx:504]: This is interesting - makes each canvas element tabbable, though as a user when pressing "enter" I might expect that action would "select" the currently tabbed element 🤔 (which is what click does) Maybe we should expand the onElementKeyDown function to also select the current tabbed selection?

**Comment by nmarrs** [comment]
We discussed today that a "good enough" step 1 of this is supporting making canvas elements tabbable with the "enter" key selecting the element in this PR.

Supporting tooltip / datalink functionality is a little trickier, and it something we will need to consider in the [tooltip epic](https://github.com/grafana/grafana/issues/68238). We should add a github issue to the "phase 2" of that project for an accessibility overview / one specific for panels like geomap / canvas

File: public/app/features/canvas/runtime/element.tsx (hunk 1 above)
Comment (nmarrs) [inline public/app/features/canvas/runtime/element.tsx:504]: maybe something like?

```suggestion
onElementKeyDown = (event: React.KeyboardEvent) => {
    if (event.key === 'Enter' && (event.currentTarget instanceof HTMLElement || event.currentTarget instanceof SVGElement)) {
      const scene = this.getScene();
      scene?.select({targets: [event.currentTarget]})
    }
  };
```

Comment (Develer) [reply public/app/features/canvas/runtime/element.tsx]: Thank you for the types solution! It works! 💙
Unfortunately, I didn't find a more elegant solution here 🙁

**Review by nmarrs — APPROVED** [review]
LGTM

**Pull request merged**